### ISSUE TITLE
Aliasing

### ISSUE THREAD

**Issue by cyriltasse** [ratt-ru/bullseye#60]
I've noted a simple issue, that is probably absolutely normal, but I did not give any thought about it before. If you make an image with and off-field source, one will see sidelobes across the image, but also some additional noise (plus some fixed ghost sources) that I attribute to alising. I think that is the issue because if you increase the FOV, this noise decrease (we basically decrease the size of the cells in the uv-space). That is the case for my gridder, but also the case for casa. This is somewhat a problem since the smaller the facet, the more we'll have this issue. Do you also observe that with your gridder?

The other strange thing is that I think the degridding should not be affected (think of a dataset with only one source in one facet only). So clean still does properly converge even if facets are small.

I've been thinking about the problem but could not find a proper way to look at it and fully understand it.

**Comment by bennahugo** [comment]
Hi Cyril. Yes this aliasing (if it is aliasing) is an issue across the board it seems -ghosts clearly visible. I've checked that this is not due to subsampling the anti-aliasing filter. Not so sure why the noise will increase with smaller facets? The aliases (near field centre) are still there after I either: increased the cell size (arcsec) keeping number pixels fixed or kept the cell size fixed and increased the number of pixels. The setup are as follows:

python bullseye/bullseye.py "/scratch/test/widefield.ms" --output_prefix /scratch/test/out8arcsec_quarterres --npix_l 256 --npix_m 256 --cell_l 8 --cell_m 8 --pol XX --output_format fits --field_id 0 --data_column "DATA" --average_all 1 --open_default_viewer 1 --wplanes 1 --conv_sup 20 --conv_oversamp 50 --no_chunks 50

I've made some maps for you, including a "Nearest Neighbour" interpolation (ie. no anti-aliasing) to show that the filter is indeed reducing aliases - can't attach them here zip is too big - emailed you instead. It may just be that the sidelobes are suffering some sort of maximum destructive interference at these points

**Comment by bennahugo** [comment]
Oh yea: that is using an ordinary sinc function with a very very big support region

**Comment by cyriltasse** [comment]
Thanks, another way to setup the experiment, for those not following the discussion:

- Make the image in one facet only, the source being *outside* the facet the FOV, say at twice the facet diameter.
- Now take the same data, and include that source in your facet (just increase the number of pixels).
- If you compare the "large" image with the "small" image at the overlap you'll see that the small one has an additional noise.

**Comment by o-smirnov** [comment]
CASA has a padding argument (or at least lwimager does) for precisely this
reason. I think it just zero-pads the uv-grid.

**Comment by cyriltasse** [comment]
For me the padding is to avoid for extremely high/low values at the edge of the field when normalising by the spheroidal in the image domain after gridding. So do your gridding, fft, normalise, then cut out the edge where it's crap. Of course you're right, it will also reduce the aforementioned effect too. However, to completely cancel this noise that way, padding should not just be 1.2 as is used in casa, but should correspond to the FOV that include the most distant correlated source [times PB], derived from dnu-dt, well you know this by heart :)

So if that's a problem (I'm not sure it is since degridding I think works), and what you say is right, facets can't be useful since facets have to have the size of the primary beam basically.

**Comment by bennahugo** [comment]
Cyril is there any reason why you can't pad the facets and after the fft crop them to whatever size you want just like any other image? The energy of the sources towards the edge of the padded image will be smeared for sure, but their total energy over the entire region should remain constant? I've emailed you my images for the experiment you suggested. After equalizing the two images the beams at the centre of the image appear to be exactly the same (except towards the edges as expected

**Comment by bennahugo** [comment]
Upon close inspection I notice there is a very tiny discrepancy between the pixel values. This may very well be a floating point issue. I've also emailed you double precision gridded and FFTed results.

One theory:
If this is the kind of noise you're hunting then I believe we should grid in double precision anyway (perhaps including a reduction step to avoid adding noise-like visibilities to continuously increasing grid cells).
 
Another (or in combination with):
Some of the visibilities fall off the edge of the grid (either completely or some of the convolution filter does). This will have a very real impact on the total energy in the final image. As Oleg suggested pad the grid (total grid resolution twice maximum u and v respectively). However, this means each facet is as large as the original w-projected image (in terms of memory)...

**Comment by cyriltasse** [comment]
Hey benna, thanks for the tests. I should have mentioned I'm weighting the visibilities with Briggs scheme and a low robust parameter (-1), so almost in uniform. When weighting in natural, I almost don't see anything significant. So which weighting scheme do you use, I guess natural? It's not even obvious to me why that is only significant for the longest baselines, maybe something related to Oleg's deamon: the "size" of the visibility. Now I'm talking about it, I'm generating a visibility every 30 seconds (lofar LBA), maybe I can try to increase the sample rate, and see how that goes...

On your side, can you do briggs, or uniform? If not, can you send me your MS? And I'll try to reproduce your two images and compare...

**Comment by bennahugo** [comment]
Yup it is natural weighting...  haven't used my uniform weighting in a very long time... seems very broken... will fix. Don't have an option to do Briggs I'm afraid. See my email.
